web_fragments/tests/test_fragment.py

```diff
@@ -0,0 +1,264 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+"""
+Unit tests for the Fragment class.
+"""
+
+from __future__ import absolute_import, unicode_literals
+
+import ddt
+
+from django.test import TestCase
+
+from web_fragments.fragment import Fragment, FragmentResource
+from web_fragments.test_utils import (
+    CSS_EXPECTED_HTML, CSS_LINK_EXPECTED_HTML, EXPECTED_JS_INIT_VERSION, JS_EXPECTED_HTML,
+    JS_LINK_EXPECTED_HTML, TEST_CSS, TEST_CSS_URL, TEST_HTML, TEST_JS, TEST_JS_INIT_FN,
+    TEST_JS_URL, TEST_JSON_INIT_ARGS
+)
+
+
+@ddt.ddt
+class TestFragment(TestCase):
+    """
+    Unit tests for fragments.
+    """
+
+    def setUp(self):
+        super(TestFragment, self).setUp()
+
+    def create_test_fragment(self):
+        """
+        Creates a fragment for use in unit tests.
+        """
+        fragment = Fragment()
+        fragment.add_content(TEST_HTML)
+        fragment.add_css(TEST_CSS)
+        fragment.add_css_url(TEST_CSS_URL)
+        fragment.add_javascript(TEST_JS)
+        fragment.add_javascript_url(TEST_JS_URL)
+        fragment.initialize_js(TEST_JS_INIT_FN, json_args=TEST_JSON_INIT_ARGS)
+        return fragment
+
+    def validate_fragment(self, fragment=None, fragment_dict=None):
+        """
+        Validates that the fields of a fragment are all correct.
+        """
+        fragment_dict = fragment_dict if fragment_dict else fragment.to_dict()
+        self.assertEqual(fragment_dict['content'], TEST_HTML)
+        self.assertEqual(fragment_dict['js_init_fn'], TEST_JS_INIT_FN)
+        self.assertEqual(fragment_dict['js_init_version'], EXPECTED_JS_INIT_VERSION)
+        self.assertEqual(fragment_dict['json_init_args'], TEST_JSON_INIT_ARGS)
+        self.assertEqual(fragment_dict['resources'], self.test_resources_dict())
+
+    def test_resources_dict(self):
+        """
+        Returns the expected test resources.
+        """
+        return [
+            {
+                'kind': 'text',
+                'data': TEST_CSS,
+                'mimetype': 'text/css',
+                'placement': 'head',
+            },
+            {
+                'kind': 'url',
+                'data': TEST_CSS_URL,
+                'mimetype': 'text/css',
+                'placement': 'head',
+            },
+            {
+                'kind': 'text',
+                'data': TEST_JS,
+                'mimetype': 'application/javascript',
+                'placement': 'foot',
+            },
+            {
+                'kind': 'url',
+                'data': TEST_JS_URL,
+                'mimetype': 'application/javascript',
+                'placement': 'foot',
+            },
+        ]
```

Comment:
Come to think of it, this isn't a `dict` at all, as the name implies. It's a `list`. The name is doubly misleading.